### DISCUSSION TITLE
Patient process

### DISCUSSION THREAD

**Discussion by mosesmbadi** [mosesmbadi/easymed#67] [General]
THIS IS LEGACY. SEE #628  FOR MORE DETAILS
@Jaydrey 
There are a few observations I've noted. 
A doctor is assigned to an appointment, not a patient. During the whole Patient's visit, they might be required to see different doctors/specialists. Therefore we assign a doctor to a process, the same way we will need to assign a LabTech to a LabTest request/results and not to the patient. This essentially means, the assign-doctor an appointment will be a PUT request to the Appointment table.
The endpoint in question is this one;
```
class AssignPatientToDoctorAPIView(APIView):

    @extend_schema(
        request =AssignPatientToDoctorSerializer,
    )
    def post(self, request: Request, *args, **kwargs):
        data: dict = request.data
        serializer = AssignPatientToDoctorSerializer(data=data)
        serializer.is_valid(raise_exception=True)
        message = serializer.assign_doctor()
        return Response({"message": message}, status=status.HTTP_200_OK)
```

Converting a Public appointment to a patient/appointment should be a POST request to either the appointments table or patients table.

What is the difference between these two classes;
```
class CreatePatientAPIView(APIView):
    permission_classes = (AllowAny,)
    @extend_schema(
        request =CreatePatientSerializer,
    )
    def post(self, request: Request, *args, **kwargs):
        data = request.data
        serializer = CreatePatientSerializer(data=data)
        serializer.is_valid(raise_exception=True)
        message = serializer.create_patient()
        return Response(message, status=status.HTTP_201_CREATED)

class PatientViewSet(viewsets.ModelViewSet):
    queryset = Patient.objects.all()
    serializer_class = PatientSerializer
```

The SystemAdmin endpoints should be accessible to Django's SuperUser user role too.

Just to clarify, the system has Users, patients, Doctors, LabTechs, Nurses, 
Doctors, Nurses, LabTechs and any other hospital staff will be a user who has those respective roles.
A patient in the system may or may not be a user. There are patients who just walk into the hospital, they are registered and their processes continue from there i.e booking appointments. There are patients who will register on the landing page from their homes, and then book a session there. When they physically come to the hospital, they will not be registered again, the receptionist will ask for their name, check in the system then book an appointment and assign a doctor. There are also patients who don't want to register as a user, they just book an appointment from their homes, which goes to the PublicAppointments. In this regard, you have two types of appointments, PubLicAppointments (from patients who are not users nor registered patients) and Appointments (from patients who are registered as patients and users).

**Comment by Jaydrey** [comment]
@mosesmbadi You are right! The doctor is assigned a process. I earlier on thought , a doctor is assigned a patient but I saw how the appointment table looked like and I also arrived to the same conclusion. 

So in the AssignPatientToDoctorAPIView, we pass the doctor id and the appointments id, of the Appointment Table. That way a receptionist can assign the doctor multiple processes if it makes sense. 

From the summary you've made below, it makes sense now. I don't think I'll have to change anything in the models then. I'll work with what is there. I'll probably just add the VisitHistory model to track the visists and stages the patient is currently on.
